directjngine/src/main/java/com/softwarementors/extjs/djn/servlet/DirectJNgineServlet.java

```diff
@@ -657,4 +676,31 @@ private void processUploadFormPost(HttpServletRequest request, HttpServletRespon
     return uploader.parseRequest(request);
   }
 
+  private boolean isValidAntiCsrfTokenMatching(final HttpServletRequest request, final List<FileItem> fileItems) {
+    String cookie = getAntiCsrfTokenCookie(request);
+    String form = getAntiCsrfTokenField(fileItems);
+
+    return !antiCsrfTokenEnabled || (cookie != null && cookie.equals(form));
+  }
+
+  private String getAntiCsrfTokenCookie(final HttpServletRequest request) {
+    Cookie[] cookies = request.getCookies();
+    if (cookies != null) {
+      for (Cookie cookie : cookies) {
+        if (ANTI_CSRF_TOKEN_NAME.equals(cookie.getName())) {
+          return cookie.getValue();
+        }
+      }
+    }
+    return null;
+  }
+
+  private String getAntiCsrfTokenField(final List<FileItem> fileItems) {
+    for (FileItem item : fileItems) {
```

Comment:
Is Java 8 available in this project? We could use a stream instead.

```
return fileItems.stream()
  .filter(FileItem::isFormField)
  .filter(item -> ANTI_CSRF_TOKEN_NAME.equals(item.getFieldName()))
  .findAny().orElse(null);
```

Reply:
This project is running on Java 1.5 unfortunately...